contracts/trading/seaport16/zones/immutable-signed-zone/v3/ImmutableSignedZoneV3.sol

```diff
@@ -0,0 +1,834 @@
+        // (strictly requires 64 byte compact sig, ERC2098).
+        bytes calldata signature = extraData[29:93];
+
+        // extraData bytes 93-end: context (optional, variable length).
+        bytes calldata context = extraData[93:];
+
+        // Revert if expired.
+        // solhint-disable-next-line not-rely-on-time
+        if (block.timestamp > expiration) {
+            // solhint-disable-next-line not-rely-on-time
+            revert SignatureExpired(block.timestamp, expiration, orderHash);
+        }
+
+        // Put fulfiller on the stack for more efficient access.
+        address actualFulfiller = zoneParameters.fulfiller;
+
+        // Revert unless:
+        // - expected fulfiller is 0 address (any fulfiller) OR
+        // - expected fulfiller is the same as actual fulfiller.
+        if (expectedFulfiller != address(0) && expectedFulfiller != actualFulfiller) {
+            revert InvalidFulfiller(expectedFulfiller, actualFulfiller, orderHash);
+        }
+
+        // Validate supported substandards - before hook validation.
+        _validateSubstandards(context, zoneParameters, true);
+
+        // Derive the signedOrder hash.
+        bytes32 signedOrderHash = _deriveSignedOrderHash(expectedFulfiller, expiration, orderHash, context);
+
+        // Derive the EIP-712 digest using the domain separator and signedOrder
+        // hash through openzepplin helper.
+        bytes32 digest = MessageHashUtils.toTypedDataHash(_domainSeparator(), signedOrderHash);
+
+        // Recover the signer address from the digest and signature.
+        // Pass in R and VS from compact signature (ERC2098).
+        address recoveredSigner = ECDSA.recover(digest, bytes32(signature[0:32]), bytes32(signature[32:64]));
+
+        // Revert if the signer is not active.
+        // This also reverts if the digest constructed on serverside is incorrect.
+        if (!_signers[recoveredSigner].active) {
+            revert SignerNotActive(recoveredSigner);
+        }
+
+        // Pre hook validation completes and passes with no reverts, return valid.
+        authorizedOrderMagicValue = ZoneInterface.authorizeOrder.selector;
+    }
+
+    /**
+     * @notice Validates a fulfilment execution.
+     *
+     * @dev This function is called by Seaport whenever any extraData is
+     *      provided by the caller, after tokens have been transferred.
+     *
+     * @param zoneParameters        The zone parameters containing data related to
+     *                              the fulfilment execution.
+     * @return validOrderMagicValue A magic value indicating if the order is
+     *                              currently valid.
+     */
+    function validateOrder(
+        ZoneParameters calldata zoneParameters
+    ) external override returns (bytes4 validOrderMagicValue) {
+        // Put the extraData and orderHash on the stack for cheaper access.
+        bytes calldata extraData = zoneParameters.extraData;
+
```

Comment:
Please add this check so a meaningful error is produced if the length is incorrect.

        // Revert with an error if the extraData does not have valid length.
        if (extraData.length < 93) {
            revert InvalidExtraDataLength("extraData length must be at least 93 bytes", orderHash, extraData.length);
        }

Reply:
The `validateOrder` function intentionally avoids redundant checks already validated as part of the `authorizeOrder` function (which is guaranteed to execute before `validateOrder` during order fulfilment) to reduce overhead.